src/client.ts

```diff
@@ -55,6 +60,7 @@ interface AdditionalOptions {
   userAgent?: string;
   includeExplanation?: boolean;
   apiKey?: string;
+  logger?: LoggerFunc;
```

Comment:
Since we only want to expose a single place for the logger to be passed in (during Client construction), we shouldn't expose this via `AdditionalOptions` as well.

It should be moved from here to the constructors for each of the three sub-clients, and then stored as private variables in those classes.

Reply:
Updated